### PR TITLE
Allows drag of autohide window

### PR DESCRIPTION
**Pull request by zii-dmg**

### CODE REVIEW

File: WinFormsUI/Docking/DockPaneCaptionBase.cs [hunk 1]
```diff
@@ -55,7 +55,6 @@ protected override void OnMouseDown(MouseEventArgs e)
             if (e.Button == MouseButtons.Left &&
                 DockPane.DockPanel.AllowEndUserDocking &&
                 DockPane.AllowDockDragAndDrop &&
-                !DockHelper.IsDockStateAutoHide(DockPane.DockState) &&
```
Comment (lextm) [inline WinFormsUI/Docking/DockPaneCaptionBase.cs]: If you change this, then all themes are affected. But personally I think VS2003 and VS2005 themes should not be affected.

Can you simply override `OnMouseDown` in VS2012/VS2013/VS2015 themes for captions and make the necessary changes?

**Comment by zii-dmg** [comment]
Don't see a way to just override OnMouseDown, because all logic is in base method. Without calling base method we don't have event MouseDown.

We can introduce new virtual property:
```cs
class DockPaneCaptionBase {
    protected virtual bool CanDragAutoHide { get; }
}
```
Override it in themes panes and add check in base `OnMouseDown`.

What do you think?

**Comment by lextm** [comment]
Please go ahead and implement the property. I think it is more practical.

**Comment by lextm** [comment]
Thanks. Merging now.

**Pull request merged**